synapseclient/api/configuration_services.py

```diff
@@ -0,0 +1,136 @@
+"""This module is responsible for exposing access to any configuration either through
+file, environment variables, or other means.
+"""
+
+import configparser
+import functools
+from typing import Dict
+
+from synapseclient.core.constants import config_file_constants
+from synapseclient.core.pool_provider import DEFAULT_NUM_THREADS
+
+
+@functools.lru_cache()
+def get_config_file(config_path: str) -> configparser.RawConfigParser:
+    """
+    Retrieves the client configuration information.
+
+    Arguments:
+        config_path:  Path to configuration file on local file system
+
+    Returns:
+        A RawConfigParser populated with properties from the user's configuration file.
+    """
+
+    try:
+        config = configparser.RawConfigParser()
+        config.read(config_path)  # Does not fail if the file does not exist
+        return config
+    except configparser.Error as ex:
+        raise ValueError(f"Error parsing Synapse config file: {config_path}") from ex
+
+
+def get_config_section_dict(
+    section_name: str,
+    config_path: str,
+) -> Dict[str, str]:
```

Comment:
I think return type should be `Dict[str, Any]`

Reply:
Each of the configuration sections are simple key,value pairs. The config parser library that is extracting these values is always returning back a string for the value even if it's not exactly a string:

![image](https://github.com/Sage-Bionetworks/synapsePythonClient/assets/17128019/feb2c7c8-96fc-45ef-9d25-32c54ed486ed)


![image](https://github.com/Sage-Bionetworks/synapsePythonClient/assets/17128019/59b11a7d-5218-4fae-a8b7-f3174bf5a834)